File: mod/src/client/java/top/fifthlight/armorstand/extension/RenderPipelineExt.java
package top.fifthlight.armorstand.extension;

import org.jetbrains.annotations.NotNull;
import top.fifthlight.armorstand.model.VertexType;

import java.util.List;
import java.util.Optional;

public interface RenderPipelineExt {
    @NotNull
    List<String> armorStand$getUniformBuffers();

    @NotNull
    Optional<VertexType> armorStand$getVertexType();
}

File: mod/src/client/kotlin/top/fifthlight/armorstand/extension/RenderPipelineExt.kt
package top.fifthlight.armorstand.extension

import com.mojang.blaze3d.pipeline.RenderPipeline

val RenderPipeline.uniformBuffers
    get() = (this as RenderPipelineExt).`armorStand$getUniformBuffers`()

val RenderPipeline.vertexType
    get() = (this as RenderPipelineExt).`armorStand$getVertexType`()
